### ISSUE TITLE
`needless_borrow` does not warn on needless `&mut` borrows

### ISSUE THREAD

**Issue by lopopolo** [rust-lang/rust-clippy#7635]
Lint name:

I tried this code:

```rust
#![deny(clippy::needless_borrow)]

fn take_borrow(p: &i64) {
    println!("{}", *p);
}

fn take_mut_borrow(p: &mut i64) {
    *p = 27;
}

fn main() {
    let mut p = 1;
    take_mut_borrow(&mut p);
    take_mut_borrow(&mut &mut p);
    take_borrow(&p);
    take_borrow(&&p);
    take_borrow(& &mut p);
}
```

https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=ff47b3fc69b998a3787cba0fe34e8006

I expected to see this happen:

I expected to see warnings for:

- `take_mut_borrow(&mut &mut p);`
- `take_borrow(&&p);`
- `take_borrow(& &mut p);`

Instead, this happened:

I only saw warnings for:

- `take_borrow(&&p);`
- `take_borrow(& &mut p);`

### Meta

**Rust version (`rustc -Vv`):**

Stable channel
Build using the Stable version: 1.54.0
